New color guides and some associated link updates (#33071)
mdn/content: files/en-us/web/css/css_colors/applying_color/index.md

---
title: Applying color to HTML elements using CSS
short-title: Applying color to HTML elements
slug: Web/CSS/CSS_colors/Applying_color
page-type: guide
---

{{CSSRef}}

With [CSS](/en-US/docs/Web/CSS), there are lots of ways to add color to your [HTML](/en-US/docs/Web/HTML) [elements](/en-US/docs/Web/HTML/Element) to create the look you want. This guide is a primer introducing how CSS can be used to apply colors to HTML elements. This guide includes [lists of the CSS properties that set color in their values](#properties_that_can_have_color) and how to use colors both [in stylesheets](#specifying_colors_as_values_in_stylesheets) and [in other ways](#other_ways_to_use_color).

> > **Note:** It is important to [use colors wisely](/en-US/docs/Web/CSS/CSS_colors/Using_color_wisely). Always select appropriate colors, ensure there is enough contrast between text and the background to ensure legibility, and always keep the needs of people with differing visual capabilities in mind.

Author: went with:

> ensuring the contrast between text and the background is sufficient

## Specifying colors as values in stylesheets

Now that you know which [CSS properties let you apply color to elements](#properties_that_can_have_color), you can start adding colors to your websites. Let's look at some examples of using color within a {{Glossary("stylesheet")}}. In this example, we use several previously mentioned properties, with the concept of applying colors in CSS being the same no matter the property.

Let's look at the result first, before going on to look at the code we need to create it:

{{EmbedLiveSample("Specifying colors as values in stylesheets", 650, 150)}}

### HTML

The HTML responsible for creating the above example is shown here:

```html
<div class="wrapper">
  <div class="boxLeft">
    <p>This is the first box.</p>
  </div>
  <div class="boxRight">
    <p>This is the second box.</p>
  </div>
</div>
```

Here we have a wrapper {{HTMLElement("div")}} containing two child `<div>`s, each with a single child paragraph ({{HTMLElement("p")}}). Each content `<div>` is given a different look and feel.

### CSS

Let's look at the CSS that creates the above result a piece at a time.

> > **Note:** We are using multiple [different CSS color notations](/en-US/docs/Web/CSS/CSS_colors/Color_values) in this example to demonstrate their use. This is not recommended for production code. When writing CSS, use the style of writing CSS colors that is most intuitive for you and your team.

Reviewer: ```suggestion
> **Note:** We are using multiple [different CSS color value types](/en-US/docs/Web/CSS/CSS_colors/Color_values) in this example to demonstrate their use. This is not recommended for production code. When writing CSS, use the most intuitive value type for you and your team.
```

Reviewer: Simplified; I also wanted to make sure the link text more closely matched the name of the page you were linking to, to avoid potential confusion.

- We add a green wavy line under the text with the {{cssxref("text-decoration")}} shorthand, along with the longhand component for browser compatibility. We used the 3-digit {{cssxref("hex-color")}} `#8f8`, which is the equivalent of `#88ff88`.
- Finally, a bit of a shadow is added to the text using {{cssxref("text-shadow")}}. Its `color` parameter is set to `black`.

## Other ways to use color

CSS isn't the only web technology that supports color. There are graphics technologies that are available on the web which also support color.

- The HTML [Canvas API](/en-US/docs/Web/API/Canvas_API)
  - : Lets you draw 2D bitmapped graphics in a {{HTMLElement("canvas")}} element. See our [Canvas tutorial](/en-US/docs/Web/API/Canvas_API/Tutorial) to learn more.
- [SVG](/en-US/docs/Web/SVG) (Scalable Vector Graphics)
  - : Lets you draw images using commands that draw specific shapes, patterns, and lines to produce an image. SVG commands are formatted as XML, and can be embedded directly into a web page or can be placed in the page using the {{HTMLElement("img")}} element, just like any other type of image.
- [WebGL](/en-US/docs/Web/API/WebGL_API)

>   - : The Web Graphics Library is an OpenGL ES-based API for drawing high-performance 2D and 3D graphics on the Web. See our [WebGL tutorial](/en-US/docs/Web/API/WebGL_API/Tutorial) to find out more.

Reviewer: Perfect.
